grpc/src/main/java/com/linecorp/armeria/client/grpc/GrpcWebUtil.java

```
@@ -88,39 +99,76 @@ public static HttpHeaders parseTrailers(ClientRequestContext ctx, HttpData respo
+                // 8th (MSB) bit of the 1st gRPC frame byte is:
+                // - '1' for trailers
+                // - '0' for data
+                //
+                // See: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md
+                if (type >> 7 == 1) {
+                    if ((type & 1) > 0) {
+                        // The compressed trailers.
+                        final ResponseHeaders responseHeaders = ctx.log().ensureAvailable(
+                                RequestLogProperty.RESPONSE_HEADERS).responseHeaders();
+                        final String grpcEncoding = responseHeaders.get(GrpcHeaderNames.GRPC_ENCODING);
+                        // We use DecompressorRegistry in ArmeriaClientCall. If ArmeriaClientCall
+                        // supports to add another decompressor, we will change this to support that too.
+                        final DecompressorRegistry registry = DecompressorRegistry.getDefaultInstance();
+                        if (grpcEncoding == null) {
+                            // grpc-encoding header is missing.
+                            return null;
+                        }
+                        final Decompressor decompressor = registry.lookupDecompressor(grpcEncoding);
+                        if (decompressor == null) {
+                            return null;
+                        }
+
+                        final ByteBuf outputBuf = ctx.alloc().compositeBuffer();
+                        boolean success = false;
+                        try (ByteBufInputStream is = new ByteBufInputStream(buf);
+                             InputStream decompressIs = decompressor.decompress(is);
+                             ByteBufOutputStream os = new ByteBufOutputStream(outputBuf)) {
+                            ByteStreams.copy(decompressIs, os);
+                            success = true;
+                        } catch (Throwable t) {
+                            if (!success) {
```

nit: Could remove `success` flag? I think we don't need to check that success is true.

---

or replace `catch (Throwable t)` with `finally`? It swallows the exception currently which doesn't sound right.

---

.. and merge this `try` block with the below `try` block?

---

.. or maybe it was intentional to swallow the exception. :thinking:

---

Yes, that was my intention to swallow the exception because this is a hack to extract the gRPC trailers. All invalid stuff will be dealt with the framer and `ArmeriaClientCall`. 😉

---

> Could remove success flag? I think we don't need to check that success is true.

`outputBuf` is used at line 157. If it's released, we can't use it. 😉

---

>  merge this try block with the below try block?

Merged. Thanks!